test/view_component/integration_test.rb

```diff
@@ -397,6 +397,14 @@ class IntegrationTest < ActionDispatch::IntegrationTest
       get "/controller_to_string_render_component"
       assert_includes response.body, "bar"
     end
+
+    test "rendering component in preview using #render_component and monkey patch disabled" do
+      old_default = ViewComponent::Base.render_monkey_patch_enabled
+      ViewComponent::Base.render_monkey_patch_enabled = false
```

Comment:
nitpick: What do you think about creating a helper method for overriding this config?

```ruby
# test_helper.rb

...

def with_render_monkey_patch_config(enabled)
  old_default = ViewComponent::Base.render_monkey_patch_enabled
  ViewComponent::Base.render_monkey_patch_enabled = enabled
  yield
ensure
  ViewComponent::Base.render_monkey_patch_enabled = old_default
end
```

And then use it here like:

```ruby
# test/view_component/integration_test.rb

...

test "rendering component ..." do
  with_render_monkey_patch_config(false) do
    get "/rails/view_components/monkey_patch_disabled_component/default"
    assert_includes response.body, "<div>hello,world!</div>"
  end
end